x-pack/plugins/apm/public/components/app/service_overview/service_overview_transactions_table/index.tsx

```diff
@@ -132,21 +150,24 @@ export function ServiceOverviewTransactionsTable({ serviceName }: Props) {
               transactionType,
               latencyAggregationType,
               transactionNames,
+              comparisonStart,
+              comparisonEnd,
             },
           },
         });
       }
     },
-    // only fetches statistics when requestId changes or transaction names changes
+    // only fetches statistics when requestId, transaction names or comparison range change
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [requestId, transactionNames],
+    [requestId, transactionNames, comparisonStart, comparisonEnd],
```

Comment:
Depending on what is agreed here https://github.com/elastic/kibana/pull/89138#discussion_r576324820, I'll refactor this.